tests/python_client/testcases/test_query.py

```diff
@@ -654,6 +640,268 @@ def test_query_expr_json_contains_combined_with_normal(self, enable_dynamic_fiel
             res = collection_w.query(expression)[0]
             assert len(res) == limit // 2
 
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_expr_all_datatype_json_contains_all(self, enable_dynamic_field):
+        """
+        target: test query with expression using json_contains
+        method: query with expression using json_contains
+        expected: succeed
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, enable_dynamic_field=enable_dynamic_field)[0]
+
+        # 2. insert data
+        array = cf.gen_default_rows_data()
+        limit = 10
+        for i in range(ct.default_nb):
+            content = {
+                "listInt": [m for m in range(i, i + limit)],  # test for int
+                "listStr": [str(m) for m in range(i, i + limit)],  # test for string
+                "listFlt": [m * 1.0 for m in range(i, i + limit)],  # test for float
+                "listBool": [bool(i % 2)],  # test for bool
+                "listList": [[i, str(i + 1)], [i * 1.0, i + 1]],  # test for list
+                "listMix": [i, i * 1.1, str(i), bool(i % 2), [i, str(i)]]  # test for mixed data
+            }
+            array[i][ct.default_json_field_name] = content
+
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expr_prefix = ["json_contains_all", "JSON_CONTAINS_ALL"]
+        for expr in expr_prefix:
+            # test for int
+            _id = random.randint(0, ct.default_nb)
+            ids = [i for i in range(_id, _id + limit)]
+            expression = f"{expr}(json_field['listInt'], {ids})"
```

Comment:
what is the behavior for json_contains_all(json_array, []) and json_contains_any(json_array, [])
Add a case to cover it.

Reply:
json_contains_all(json_array, []) has been added in case test_query_expr_json_contains_invalid_type.